mail_restrict_auto_follower/models/res_partner.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models
+
+
+class ResPartner(models.Model):
+    """This is for avoid inheritance problem:
+    https://github.com/odoo/odoo/issues/9084"""
+    _name = 'res.partner'
```

Comment:
this is specific to `res.partner` now. Can't you enumerate all models in `_register_hook` and patch the ones who need it?